pykokkos/lib/ufuncs.py

```diff
@@ -3077,3 +3077,390 @@ def isfinite(view):
                             view=view,
                             out=out)
     return out
+
+@pk.workunit
+def round_impl_1d_double(tid: int, view: pk.View1D[pk.double], out: pk.View1D[pk.double]):
+    out[tid] = round(view[tid])
+
+
+@pk.workunit
+def round_impl_2d_double(tid: int, view: pk.View2D[pk.double], out: pk.View2D[pk.double]):
+    for i in range(view.extent(1)):
+        out[tid][i] = round(view[tid][i])
+
+
+@pk.workunit
+def round_impl_3d_double(tid: int, view: pk.View3D[pk.double], out: pk.View3D[pk.double]):
+    for i in range(view.extent(1)):
+        for j in range(view.extent(2)):
+            out[tid][i][j] = round(view[tid][i][j])
+
+@pk.workunit
+def round_impl_1d_float(tid: int, view: pk.View1D[pk.float], out: pk.View1D[pk.float]):
+    out[tid] = round(view[tid])
+
+
+@pk.workunit
+def round_impl_2d_float(tid: int, view: pk.View2D[pk.float], out: pk.View2D[pk.float]):
+    for i in range(view.extent(1)):
+        out[tid][i] = round(view[tid][i])
+
+
+@pk.workunit
+def round_impl_3d_float(tid: int, view: pk.View3D[pk.float], out: pk.View3D[pk.float]):
+    for i in range(view.extent(1)):
+        for j in range(view.extent(2)):
+            out[tid][i][j] = round(view[tid][i][j])
+
+def round(view):
+    """
+    Rounds each element of the input view to the nearest integer-valued number.
+
+    Parameters
+    ----------
+    view : pykokkos view
+           Should have a numeric data type.
+
+    Returns
+    -------
+    out: pykokkos view
+         A view containing the rounded result for each element in
+         the input view. The returned view must have the same data
+         type as the input view.
+
+    Notes
+    -----
+    If view element ``i`` is already integer-valued, the result is ``i``.
+
+    """
+    if "int" in str(view.dtype):
+        # special case defined in API std
+        return view
+    out = pk.View(view.shape, dtype=view.dtype)
+    if len(view.shape) > 3:
+        raise NotImplementedError("only up to 3D views currently supported for round() ufunc.")
+        
+    if "double" in str(view.dtype):
```

Comment:
Rather than having three blocks for three types, can we have one if statement and then call a function to which we pass three functions, something along these lines
```
if "double" in str(view.dtype):
   _parallel_for(round_impl_1d_double, round_impl_2d_double, round_impl_3d_double...)
elif "float64" in str(view.dtype): 
   ...
```

Reply:
How does `_parallel_for` know how to dispatch to the individual (1D vs. 2D vs. 3D) functions? How do you write a single `_parallel_for` that is shared between different ufuncs? I don't think we can easily template to the workunits without basically writing a bunch of nested logic again. And we need a separate `_parallel_for` for each type? I dunno, it seems  like the abstraction is risky on making things simpler because the workunits don't really have metadata we can hook into for dispatching.

Reply:
I suppose we could write a dictionary with the workunits as values, and dispatch through well-named keys (types, and dimensions) perhaps. We might want to agree on how that would be designed first if we adopt something like that.